### ISSUE TITLE
Game Crashes

### ISSUE THREAD

**Issue by define-evil** [Axeryok/CocoaInput#4]
First of all, thank you for this great Mod, but I encountered some problems while using it. 

1. When I switch my input method back to English(US), the game crashes.
2. When I try to edit a sign, the game immediately crashes.

The only thing that seems to work normally is typing in the chat box.

macOS: 10.13.3
Minecraft: 1.12.2
Java: 1.8.0_151-b12

Here's a crash dump from the console.
```2018-03-06 21:40:27.267 java[4806:337629] Unknown modifier with keycode: 0
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fff75349ea9, pid=4806, tid=0x0000000000000307
#
# JRE version: Java(TM) SE Runtime Environment (8.0_151-b12) (build 1.8.0_151-b12)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.151-b12 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# C  [libobjc.A.dylib+0x6ea9]  objc_msgSend+0x29
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/Admin/Library/Application Support/minecraft/hs_err_pid4806.log
Compiled method (c1)  110111 12937       3       net.minecraft.client.particle.Particle::func_187110_a (292 bytes)
 total in heap  [0x000000010be7e150,0x000000010be80db8] = 11368
 relocation     [0x000000010be7e278,0x000000010be7e460] = 488
 main code      [0x000000010be7e460,0x000000010be7ffc0] = 7008
 stub code      [0x000000010be7ffc0,0x000000010be80188] = 456
 oops           [0x000000010be80188,0x000000010be80190] = 8
 metadata       [0x000000010be80190,0x000000010be801c0] = 48
 scopes data    [0x000000010be801c0,0x000000010be80a38] = 2168
 scopes pcs     [0x000000010be80a38,0x000000010be80d38] = 768
 dependencies   [0x000000010be80d38,0x000000010be80d48] = 16
 nul chk table  [0x000000010be80d48,0x000000010be80db8] = 112
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#```

**Comment by Axeryok** [comment]
I would like you to edit debugLevel 5.
See a gif: [https://gyazo.com/7bfc6765972fbd86cc680fbcc572c1ee](https://gyazo.com/7bfc6765972fbd86cc680fbcc572c1ee)
And please upload fml-client-latest.log with pastebin.

**Comment by define-evil** [comment]
Thanks for replying!

[JVM Crash Log](https://pastebin.com/Eawad0ri)
[FML Log](https://pastebin.com/nVDijvmV)

If I set my input method to Chinese **after** I enter the game, the game crashes when opening a sign but works fine opening the chat box.

If I set my input method to Chinese **before** I enter the game, the game works fine when opening a sign but crashes opening a chat box.

Kind of weird.

By the way, the highest debug level seems to be 4 for that the config automatically switched back to 4 when I entered 5 and saved.

**Comment by Axeryok** [comment]
I tried to make minecraft crash with standard Chinese IM and could not encounter the problems.
But I found a new problem.
When I set my input method to Chinese, minecraft crashes immediately.
I fixed this problem I found.
Problems you found are possibly solved,please download 3.1.3 .
If they are not solved,please tell me input method.

debugLevel.....Actually I forget number's mean but remember that higher number puts many debug logs.
I should abolish ambiguous "debugLevel" and introduce "debugMode(boolean)"

**Comment by define-evil** [comment]
Thanks for the patch. Everything works perfectly now!
